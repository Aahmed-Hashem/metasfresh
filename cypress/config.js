module.exports = {
  API_URL: 'http://w101.metasfresh.com:8081/rest/api',
  PLUGIN_API_URL: 'http://w101.metasfresh.com:9192/',
  WS_URL: 'http://w101.metasfresh.com:8081/stomp',
  username: '',
  password: '',
};
