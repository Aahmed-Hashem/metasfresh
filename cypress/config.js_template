module.exports = {
  API_URL: 'https://dev199.metasfresh.com/rest/api',
  PLUGIN_API_URL: 'http://w101.metasfresh.com:9192/',
  WS_URL: 'https://dev199.metasfresh.com/stomp',
  username: 'dev',
  password: 'demo1234',
};
