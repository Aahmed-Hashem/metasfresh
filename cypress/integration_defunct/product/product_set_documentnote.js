describe('Product DocumentNote test', function() {
  before(function() {
    cy.loginByForm();
  });

  it('open the product window', function() {
    cy.visit('window/140/2005577');
    cy.waitForHeader();
    cy.openAdvancedEdit();
    cy.writeIntoTextField('DocumentNote', '{selectall}{backspace}blah-blah-blah');
    cy.pressDoneButton();
  });
});
