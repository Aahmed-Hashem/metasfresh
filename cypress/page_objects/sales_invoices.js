import Metasfresh from './page';

class SalesInvoices extends Metasfresh {
  constructor() {
    super();

    this.windowId = 167;
    this.tableRows = '.table-flex-wrapper-row';
    this.rowSelector = 'tbody tr';
    this.listHeader = '.document-list-header';
    this.selectedRows = '.row-selected';
  }

  visit() {
    cy.visit(`/window/${this.windowId}`);
  }

  getRows() {
    return cy.get(this.tableRows).find(this.rowSelector);
  }
}

export const salesInvoices = new SalesInvoices();
