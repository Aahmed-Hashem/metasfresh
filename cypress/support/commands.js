// ***********************************************
// This example commands.js shows you how to
// create various custom commands and overwrite
// existing commands.
//
// For more comprehensive examples of custom
// commands please read more here:
// https://on.cypress.io/custom-commands
// ***********************************************
//
//
// -- This is a parent command --
// Cypress.Commands.add("login", (email, password) => { ... })
//
//
// -- This is a child command --
// Cypress.Commands.add("drag", { prevSubject: 'element'}, (subject, options) => { ... })
//
//
// -- This is a dual command --
// Cypress.Commands.add("dismiss", { prevSubject: 'optional'}, (subject, options) => { ... })
//
//
// -- This is will overwrite an existing command --
// Cypress.Commands.overwrite("visit", (originalFn, url, options) => { ... })

import { List } from 'immutable';
import { goBack, push } from 'react-router-redux';

import { loginSuccess } from '../../src/actions/AppActions';
import Auth from '../../src/services/Auth';

context('Reusable "login" custom command', function() {
  Cypress.Commands.add('loginByForm', (username, password, redirect) => {
    Cypress.log({
      name: 'loginByForm',
      message: username + ' | ' + password,
    });

    const handleSuccess = function(){
      if (redirect) {
        Cypress.reduxStore.dispatch(goBack());
      } else {
        Cypress.reduxStore.dispatch(push('/'));
      }
    };

    const checkIfAlreadyLogged = function() {
      const error = new Error('Error when checking if user logged in')

      return cy.request({
        method: 'GET',
        url: 'http://w101.metasfresh.com:8081/rest/api/login/isLoggedIn',
        failOnStatusCode: false,
        followRedirect: false,
      }).then(response => {
        if (!response.body.error) {
          return Cypress.reduxStore.dispatch(push('/'));
        }

        return Promise.reject(error);
      });
    };

    const auth = new Auth();

    cy.on("emit:reduxStore", store => {
      Cypress.reduxStore = store;
    });

    cy.visit('/login');

    return cy.request({
      method: 'POST',
      url: 'http://w101.metasfresh.com:8081/rest/api/login/authenticate',
      failOnStatusCode: false,
      followRedirect: false,
      body: {
        username,
        password,
      },
    }).then(response => {
      if (!response.isOkStatusCode) {
        return checkIfAlreadyLogged();
      }

      if (response.body.loginComplete) {
        return handleSuccess();
      }
      const roles = List(response.body.roles);

      return cy.request({
        method: 'POST',
        url: 'http://w101.metasfresh.com:8081/rest/api/login/loginComplete',
        body: { ...roles.get(0) },
        failOnStatusCode: false,
      }).then(() => {
        Cypress.reduxStore.dispatch(loginSuccess(auth));
        
        handleSuccess();
      });
    })
  });
});
