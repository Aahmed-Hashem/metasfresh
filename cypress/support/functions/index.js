const toggleFrequentFilters = () => {
  cy.clickElementWithClass('.filters-frequent .toggle-filters');
  cy.get('.filter-widget').should('exist');
};

const toggleNotFrequentFilters = () => {
  cy.clickElementWithClass('.filters-not-frequent .toggle-filters');
  cy.get('.filter-menu').should('exist');
};

const selectFrequentFilterWidget = () => {
  return cy.get('.filters-frequent .filter-widget');
};

const selectNotFrequentFilterWidget = filterId => {
  cy.clickElementWithClass(`.filter-menu .filter-option-${filterId}`, true);
  return cy.get('.filters-not-frequent .filter-widget');
};

const applyFilters = () => {
  cy.clickElementWithClass('.filter-btn-wrapper .applyBtn', true);
};

export {
  toggleFrequentFilters,
  toggleNotFrequentFilters,
  selectFrequentFilterWidget,
  selectNotFrequentFilterWidget,
  applyFilters,
};
