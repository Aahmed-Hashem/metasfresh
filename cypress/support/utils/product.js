export class Product {
  constructor(name) {
    cy.log(`Product - set name = ${name}`);
    this.name = name;
    this.attributeValues = [];
  }

  setName(name) {
    cy.log(`Product - set name = ${name}`);
    this.name = name;
    return this;
  }

  setValue(value) {
    cy.log(`Product - set value = ${value}`);
    this.value = value;
    return this;
  }

  setDescription(description) {
    cy.log(`Product - set description = ${description}`);
    this.description = description;
    return this;
  }

  setStocked(isPurchased) {
    cy.log(`Product - set isPurchased = ${isPurchased}`);
    this.isPurchased = isPurchased;
    return this;
  }

  setSold(isSold) {
    cy.log(`Product - set isSold = ${isSold}`);
    this.isSold = isSold;
    return this;
  }

  setDiverse(isDiverse) {
    cy.log(`Product - set isDiverse = ${isDiverse}`);
    this.isDiverse = isDiverse;
    return this;
  }

  setProductCategory(m_product_category) {
    cy.log(`Product - set productCategory = ${m_product_category}`);
    this.m_product_category = m_product_category;
    return this;
  }

  setProductType(productType) {
    cy.log(`Product - set productType = ${productType}`);
    this.productType = productType;
    return this;
  }

  setUOM(c_uom) {
    cy.log(`Product - set c_uom = ${c_uom}`);
    this.c_uom = c_uom;
    return this;
  }

  apply() {
    cy.log(`Product - apply - START (name=${this.name})`);
    applyProduct(this);
    cy.log(`Product - apply - END (name=${this.name})`);
    return this;
  }
}

function applyProduct(product) {
  describe(`Create new Product ${product.name}`, function() {
    cy.visitWindow('140', 'NEW');
    cy.writeIntoStringField('Name', product.name);

    // Value is readonly
    //cy.clearField('Value');
    //cy.writeIntoStringField('Value', product.value);

    cy.selectInListField('M_Product_Category_ID', product.m_product_category);

    cy.writeIntoStringField('Description', product.description);

    cy.isChecked('IsStocked').then(isIsStockedValue => {
      if (product.isStocked && !isIsStockedValue) {
        cy.clickOnCheckBox('IsStocked');
      }
    });
    cy.isChecked('IsPurchased').then(isPurchasedValue => {
      if (product.isPurchased && !isPurchasedValue) {
        cy.clickOnCheckBox('IsPurchased');
      }
    });
    cy.isChecked('IsSold').then(isSoldValue => {
      if (product.isSold && !isSoldValue) {
        cy.clickOnCheckBox('IsSold');
      }
    });
    cy.isChecked('IsDiverse').then(isDiverseValue => {
      if (product.isDiverse && !isDiverseValue) {
        cy.clickOnCheckBox('IsDiverse');
      }
    });

    cy.getFieldValue('ProductType').then(productType => {
      if (product.productType != productType) {
        cy.selectInListField('ProductType', product.productType);
      }
    });

    cy.getFieldValue('C_UOM_ID').then(uomValue => {
      if (product.c_uom && product.c_uom != uomValue) {
        cy.selectInListField('C_UOM_ID', product.c_uom);
      }
    });

    if (product.prices.length > 0) {
      product.prices.forEach(function(product) {
        applyProductPrice(product);
      });
      cy.get('table tbody tr').should('have.length', product.prices.length);
    }
  });
}

export class ProductCategory {
  constructor(name) {
    cy.log(`Product Category - set name = ${name}`);
    this.name = name;
  }

  setName(name) {
    cy.log(`Product Category - set name = ${name}`);
    this.name = name;
    return this;
  }

  setValue(value) {
    cy.log(`Product Category - set value = ${value}`);
    this.value = value;
    return this;
  }

  apply() {
    cy.log(`Product Category - apply - START (name=${this.name})`);
    applyProductCategory(this);
    cy.log(`Product Category - apply - END (name=${this.name})`);
    return this;
  }
}

function applyProductCategory(productCategory) {
  describe(`Create new Product ${productCategory.name}`, function() {
    cy.visitWindow('144', 'NEW');
    cy.writeIntoStringField('Name', productCategory.name);

    // Value is updateable
    cy.writeIntoStringField('Value', productCategory.value);
  });
}

function applyProductPrice(price) {
  describe(`Create new Product Price ${price.m_pricelist_version}`, function() {
    cy.get('#tab_M_ProductPrice').click();
    cy.pressAddNewButton();
    cy.writeIntoLookupListField(
      'M_PriceList_Version_ID',
      price.m_pricelist_version,
      price.m_pricelist_version,
      false /*typeList*/,
      true /*modal*/
    );

    cy.writeIntoStringField('PriceStd', price.priceStd, false /*modal*/, null /*rewriteUrl*/, true /*noRequest*/);

    // @TODO This approach doesn't work correctly
    // cy.writeIntoStringField('PriceStd', `{selectall}{backspace}${price.priceStd}`);

    cy.selectInListField('C_TaxCategory_ID', price.c_taxcategory);
    cy.pressDoneButton();
  });
}
