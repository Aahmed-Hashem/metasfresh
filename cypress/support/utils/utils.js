/*
 * Because often times we have to test for different languagues and some values in our fixtures
 * may be language-dependent, we have to provide versions for different languages prepended with
 * `language key`, like de_DE. Then instead of directly fetching data from the fixtures json
 * we pass it through this function, which selects keys accordingly.
 */
const getLanguageSpecific = (data, key) => {
  const lang = Cypress.reduxStore.getState().appHandler.me.language.key;

  if (lang !== 'en_US') {
    key = `${lang}__${key}`;
  }

  return data[key];
};

export { getLanguageSpecific };
