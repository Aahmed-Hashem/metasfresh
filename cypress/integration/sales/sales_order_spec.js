import { getBreadcrumbs } from '../../support/apiRequests';
import { salesOrders } from '../../page_objects/sales_orders';
import config from '../../config';
import { Product, ProductCategory } from '../../support/utils/product';
import { SalesOrder } from '../../support/utils/sales_order';
import { toggleNotFrequentFilters, selectNotFrequentFilterWidget, applyFilters } from '../../support/functions';

describe('New sales order test', function() {
  const windowId = salesOrders.windowId;
  let headerCaption = '';
  let menuOption = '';
  const timestamp = new Date().getTime();

  before(function() {
    getBreadcrumbs(windowId, '1000040-new').then(({ option, caption }) => {
      menuOption = option;
      headerCaption = caption;
    });

    const productName = `P002737_Convenience Salat 250g ${timestamp}`;
    const productValue = `Convencience Salat ${timestamp}`;
    const productCategoryName = `ProductCategoryName ${timestamp}`;
    const productCategoryValue = `ProductNameValue ${timestamp}`;

    cy.fixture('product/simple_productCategory.json').then(productCategoryJson => {
      Object.assign(new ProductCategory(), productCategoryJson)
        .setName(productCategoryName)
        .setValue(productCategoryValue)
        .apply();
    });

    cy.fixture('product/simple_product.json').then(productJson => {
      Object.assign(new Product(), productJson)
        .setName(productName)
        .setValue(productValue)
        .setProductCategory(productCategoryValue + '_' + productCategoryName)
        .apply();
    });

    salesOrders.visit();
  });

  describe('Create a new sales order', function() {
    before(function() {
      cy.get('.header-breadcrumb').contains('.header-item', headerCaption, { timeout: 10000 });

      const option = ~~(Math.random() * (2 - 0)) + 0;

      if (option === 0) {
        cy.get('.header-breadcrumb')
          .contains('.header-item', headerCaption)
          .click();

        cy.get('.header-breadcrumb')
          .find('.menu-overlay')
          .should('exist')
          .find('.menu-overlay-link')
          .contains(menuOption)
          .click();
      } else {
        cy.clickHeaderNav('New');
      }

      cy.get('.header-breadcrumb-sitename').should('contain', '<');
    });

    it('Fill Business Partner', function() {
      cy.writeIntoLookupListField('C_BPartner_ID', 'G', 'G0001_Test');

      cy.get('.header-breadcrumb-sitename').should('not.contain', '<');
    });

    // it('Create new Business Partner', function() {
    //   cy.writeIntoLookupListField('C_BPartner_ID', 'New', 'New Business Partner', true);
    //   cy.get('.panel-modal').should('exist');
    //   cy.writeIntoStringField('Companyname', vendorName);
    //   // cy.get('.header-breadcrumb-sitename').should('not.contain', '<');
    //   const aliasName = `addBP-${new Date().getTime()}`;
    //   const patchUrlPattern = '/rest/api/window/.*$';
    //   cy.server();
    //   cy.route('GET', new RegExp(patchUrlPattern)).as(aliasName);
    //   cy.get('.panel-modal-header')
    //     .find('.btn')
    //     .click();
    //   cy.wait(`@${aliasName}`);
    //   // form-field-C_Location_ID click
    //   // attributes-dropdown should.be.visible
    //   // C_Country_ID Deu Germany - Deutschland
    // });

    it('Fill order reference to differentiate cypress tests', function() {
      cy.writeIntoStringField('POReference', `Cypress Test ${new Date().getTime()}`);
      cy.get('.indicator-pending').should('not.exist');
    });

    it('Add new product', function() {
      const addNewText = Cypress.messages.window.addNew.caption;

      cy.get('.tabs-wrapper .form-flex-align .btn')
        .contains(addNewText)
        .should('exist')
        .click();

      cy.get('.panel-modal').should('exist');

      cy.get('.form-field-M_Product_ID')
        .find('input')
        .type('C');

      cy.get('.input-dropdown-list').should('exist');
      cy.contains('.input-dropdown-list-option', 'P002737_Convenience Salat 250g').click();
      cy.get('.input-dropdown-list .input-dropdown-list-header').should('not.exist');

      cy.get('.form-field-QtyEntered', { timeout: 12000 })
        .find('input')
        .should('not.have.value', '0');

      const aliasName = `addProduct-${new Date().getTime()}`;
      const patchUrlPattern = '/rest/api/window/.*$';
      cy.server();
      cy.route('GET', new RegExp(patchUrlPattern)).as(aliasName);

      cy.get('.panel-modal-header')
        .find('.btn')
        .click();

      cy.wait(`@${aliasName}`);
    });

    it('Add new product via Batch Entry', function() {
      const addNewText = Cypress.messages.window.batchEntry.caption;

      cy.get('.tabs-wrapper .form-flex-align .btn')
        .contains(addNewText)
        .should('exist')
        .click();

      cy.get('.quick-input-container').should('exist');
      cy.get('.quick-input-container').toMatchSnapshot('Empty Quick Inp');

      cy.writeIntoLookupListField('M_Product_ID', 'C', 'Convenience Salat');

      cy.get('.form-field-Qty')
        .click()
        .find('.input-body-container.focused')
        .should('exist')
        .find('i')
        .eq(0)
        .click();

      const aliasName = `addProduct-${new Date().getTime()}`;
      const patchUrlPattern = '/rest/api/window/.*$';
      cy.server();
      cy.route('GET', new RegExp(patchUrlPattern)).as(aliasName);

      cy.get('.form-field-Qty')
        .find('input')
        .should('have.value', '0.1')
        .type('1{enter}');

      cy.wait(`@${aliasName}`);

      cy.get('#lookup_M_Product_ID')
        .find('input')
        .should('have.value', '');
    });

    it('Change document status', function() {
      let completeActionCaption = '';
      const draftedCaption = Cypress.reduxStore.getState().windowHandler.master.data.DocStatus.value.caption;
      const docId = Cypress.reduxStore.getState().windowHandler.master.docId;

      cy.request('GET', `${config.API_URL}/window/${windowId}/${docId}/field/DocAction/dropdown`).then(response => {
        const resp = response.body;

        expect(resp).to.have.property('values');
        expect(resp.values.length).to.be.gt(0);

        for (let i = 0; i < resp.values.length; i += 1) {
          if (resp.values[i].key === 'CO') {
            completeActionCaption = resp.values[i].caption;

            break;
          }
        }

        cy.get('.form-field-DocAction')
          .find('.meta-dropdown-toggle')
          .click();

        cy.get('.form-field-DocAction')
          .find('.dropdown-status-toggler')
          .should('have.class', 'dropdown-status-open');

        cy.get('.form-field-DocAction .dropdown-status-list')
          .find('.dropdown-status-item')
          .contains(completeActionCaption)
          .click();

        cy.get('.indicator-pending', { timeout: 10000 }).should('not.exist');
        cy.get('.meta-dropdown-toggle .tag-success').should('not.contain', draftedCaption);
      });
    });
  });
});

describe('List tests', function() {
  const list = salesOrders;
  const timestamp = new Date().getTime();

  before(function() {
    const salesReference = `Cypress Test ${timestamp}`;

    cy.fixture('product/simple_product.json').then(() => {
      new SalesOrder(salesReference)
        .setBPartner('G0001_Test Kunde 1')
        .setBPartnerLocation('Testadresse 3')
        .apply();
    });

    salesOrders.visit();
    salesOrders.verifyElements();
  });

  it('Test if rows get selected/deselected properly', function() {
    list
      .getRows()
      .eq(0)
      .find('td')
      .eq(0)
      .type('{shift}', { release: false })
      .click();

    list
      .getRows()
      .eq(1)
      .find('td')
      .eq(0)
      .type('{shift}', { release: false })
      .click();

    list.getSelectedRows().should('have.length', 2);
    list.clickListHeader();
    list.getSelectedRows().should('have.length', 0);
  });

  it('Test if filtering works', function() {
    list.getRows().should('not.have.length', 0);

    toggleNotFrequentFilters();
    selectNotFrequentFilterWidget('default');
    cy.writeIntoStringField('DocumentNo', `${timestamp}`, false, null, true);

    applyFilters();

    list.getRows().should('have.length', 0);
  });
});

/*
 * Not implemented yet
 *
describe('Existing sales order', function() {
  const timestamp = new Date().getTime();
  const vendorName = `Cypress Test Partner #1 ${timestamp}`;

  before(function() {
    cy.loginByForm();

    cy.fixture('purchase/cypress_vendor.json').then(() => {
      new BPartner(vendorName).apply();
    });

    salesOrders.visit();
    salesOrders.verifyElements();
  });

  it('Fill Business Partner', function() {
    cy.writeIntoLookupListField('C_BPartner_ID', 'Cypress', 'Cypress Test Partner #1');
    //we need a fake click to close the opened dropdown
    cy.get('.header-breadcrumb-sitename').click();
    cy.writeIntoLookupListField('C_BPartner_Location_ID', 'Warsaw', 'Warsaw, Potocka 1', true);

    cy.get('.header-breadcrumb-sitename').should('not.contain', '<');
  });

});
*/
