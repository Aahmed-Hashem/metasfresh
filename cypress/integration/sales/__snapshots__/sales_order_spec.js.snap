exports[`New sales order test > Create a new sales order > Add new product via Batch Entry #0`] = `
<form class="quick-input-container">
  <div
    class="form-group row  form-field-M_Product_ID form-field-M_HU_PI_Item_Product_ID"
  >
    <div class="form-control-label col-sm-3" title="Product">
      <span class="zoom-into">Product</span>
    </div>
    <div class="col-sm-9 ">
      <div class="input-body-container">
        <span></span>
        <div
          class="input-dropdown-container lookup-wrapper input-secondary pulse-off input-mandatory"
        >
          <div id="lookup_M_Product_ID" class="raw-lookup-wrapper">
            <div class="lookup-widget-wrapper lookup-widget-wrapper-bcg">
              <div class="input-dropdown input-block">
                <div class="input-editable">
                  <input
                    type="text"
                    class="input-field js-input-field font-weight-semibold"
                    autocomplete="new-password"
                    tabindex="0"
                    placeholder="none"
                  />
                </div>
              </div>
            </div>
          </div>
          <div id="lookup_M_HU_PI_Item_Product_ID" class="raw-lookup-wrapper">
            <div
              class="lookup-widget-wrapper lookup-widget-wrapper-bcg raw-lookup-disabled"
            >
              <div class="input-dropdown input-block">
                <div class="input-editable">
                  <input
                    type="text"
                    class="input-field js-input-field font-weight-semibold"
                    autocomplete="new-password"
                    tabindex="0"
                    placeholder="none"
                  />
                </div>
              </div>
            </div>
          </div>
          <div class="input-icon input-icon-lg lookup-widget-wrapper">
            <i class="meta-icon-preview"></i>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="form-group row  form-field-Qty">
    <div class="form-control-label col-sm-3" title="Quantity">Quantity</div>
    <div class="col-sm-9 ">
      <div class="input-body-container">
        <span></span>
        <div class="input-block input-mandatory input-secondary pulse-off">
          <span
            class="react-numeric-input"
            style="position: relative; display: inline-block;"
            ><input
              type="text"
              autocomplete="new-password"
              class="input-field js-input-field"
              placeholder="none"
              tabindex="0"
              title=""
              value=""
              pattern=".*"
              style="padding-right: 3ex; box-sizing: border-box; font-size: inherit; border: 1px solid rgb(204, 204, 204); border-radius: 2px; padding-left: 4px; display: block; -webkit-appearance: none; line-height: normal;"/><b
              style="position: absolute; right: 2px; width: 2.26ex; border-color: rgba(0, 0, 0, 0.1); border-style: solid; text-align: center; cursor: default; transition: all 0.1s; background: rgba(0, 0, 0, 0.1); box-shadow: rgba(0, 0, 0, 0.1) -1px -1px 3px inset, rgba(255, 255, 255, 0.7) 1px 1px 3px inset; top: 2px; bottom: 50%; border-radius: 2px 2px 0px 0px; border-width: 1px 1px 0px;"
              ><i
                style="position: absolute; top: 50%; left: 50%; width: 0px; height: 0px; border-width: 0px 0.6ex 0.6ex; border-color: transparent transparent rgba(0, 0, 0, 0.7); border-style: solid; margin: -0.3ex 0px 0px -0.56ex;"
              ></i></b
            ><b
              style="position: absolute; right: 2px; width: 2.26ex; border-color: rgba(0, 0, 0, 0.1); border-style: solid; text-align: center; cursor: default; transition: all 0.1s; background: rgba(0, 0, 0, 0.1); box-shadow: rgba(0, 0, 0, 0.1) -1px -1px 3px inset, rgba(255, 255, 255, 0.7) 1px 1px 3px inset; top: 50%; bottom: 2px; border-radius: 0px 0px 2px 2px; border-width: 0px 1px 1px;"
              ><i
                style="position: absolute; top: 50%; left: 50%; width: 0px; height: 0px; border-width: 0.6ex 0.6ex 0px; border-color: rgba(0, 0, 0, 0.7) transparent transparent; border-style: solid; margin: -0.3ex 0px 0px -0.56ex;"
              ></i></b
          ></span>
        </div>
      </div>
    </div>
  </div>
  <div class="hint">(Press 'Enter' to add)</div>
  <button type="submit" class="hidden-up"></button>
</form>
`;

exports[`New sales order test > Create a new sales order > Add new order line via Batch Entry #0`] = `
<form class="quick-input-container">
  <div
    class="form-group row  form-field-M_Product_ID form-field-M_HU_PI_Item_Product_ID"
  >
    <div class="form-control-label col-sm-3" title="Product">
      <span class="zoom-into">Product</span>
    </div>
    <div class="col-sm-9 ">
      <div class="input-body-container">
        <span></span>
        <div
          class="input-dropdown-container lookup-wrapper input-secondary pulse-off input-mandatory"
        >
          <div id="lookup_M_Product_ID" class="raw-lookup-wrapper">
            <div class="lookup-widget-wrapper lookup-widget-wrapper-bcg">
              <div class="input-dropdown input-block">
                <div class="input-editable">
                  <input
                    type="text"
                    class="input-field js-input-field font-weight-semibold"
                    autocomplete="new-password"
                    tabindex="0"
                    placeholder="none"
                  />
                </div>
              </div>
            </div>
          </div>
          <div id="lookup_M_HU_PI_Item_Product_ID" class="raw-lookup-wrapper">
            <div
              class="lookup-widget-wrapper lookup-widget-wrapper-bcg raw-lookup-disabled"
            >
              <div class="input-dropdown input-block">
                <div class="input-editable">
                  <input
                    type="text"
                    class="input-field js-input-field font-weight-semibold"
                    autocomplete="new-password"
                    tabindex="0"
                    placeholder="none"
                  />
                </div>
              </div>
            </div>
          </div>
          <div class="input-icon input-icon-lg lookup-widget-wrapper">
            <i class="meta-icon-preview"></i>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="form-group row  form-field-Qty">
    <div class="form-control-label col-sm-3" title="Quantity">Quantity</div>
    <div class="col-sm-9 ">
      <div class="input-body-container">
        <span></span>
        <div class="input-block input-mandatory input-secondary pulse-off">
          <span
            class="react-numeric-input"
            style="position: relative; display: inline-block;"
            ><input
              type="text"
              autocomplete="new-password"
              class="input-field js-input-field"
              placeholder="none"
              tabindex="0"
              title=""
              value=""
              pattern=".*"
              style="padding-right: 3ex; box-sizing: border-box; font-size: inherit; border: 1px solid rgb(204, 204, 204); border-radius: 2px; padding-left: 4px; display: block; -webkit-appearance: none; line-height: normal;"/><b
              style="position: absolute; right: 2px; width: 2.26ex; border-color: rgba(0, 0, 0, 0.1); border-style: solid; text-align: center; cursor: default; transition: all 0.1s ease 0s; background: rgba(0, 0, 0, 0.1); box-shadow: rgba(0, 0, 0, 0.1) -1px -1px 3px inset, rgba(255, 255, 255, 0.7) 1px 1px 3px inset; top: 2px; bottom: 50%; border-radius: 2px 2px 0px 0px; border-width: 1px 1px 0px;"
              ><i
                style="position: absolute; top: 50%; left: 50%; width: 0px; height: 0px; border-width: 0px 0.6ex 0.6ex; border-color: transparent transparent rgba(0, 0, 0, 0.7); border-style: solid; margin: -0.3ex 0px 0px -0.56ex;"
              ></i></b
            ><b
              style="position: absolute; right: 2px; width: 2.26ex; border-color: rgba(0, 0, 0, 0.1); border-style: solid; text-align: center; cursor: default; transition: all 0.1s ease 0s; background: rgba(0, 0, 0, 0.1); box-shadow: rgba(0, 0, 0, 0.1) -1px -1px 3px inset, rgba(255, 255, 255, 0.7) 1px 1px 3px inset; top: 50%; bottom: 2px; border-radius: 0px 0px 2px 2px; border-width: 0px 1px 1px;"
              ><i
                style="position: absolute; top: 50%; left: 50%; width: 0px; height: 0px; border-width: 0.6ex 0.6ex 0px; border-color: rgba(0, 0, 0, 0.7) transparent transparent; border-style: solid; margin: -0.3ex 0px 0px -0.56ex;"
              ></i></b
          ></span>
        </div>
      </div>
    </div>
  </div>
  <div class="hint">(Press 'Enter' to add)</div>
  <button type="submit" class="hidden-up"></button>
</form>
`;
